sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/scan/OwaspZapScanner.java

```diff
@@ -0,0 +1,756 @@
+        }
+        if (Files.isDirectory(reportFile)) {
+            throw new ZapWrapperRuntimeException("For scan: " + scanContext.getContextName() + ". Report file must not be a directory!",
+                    ZapWrapperExitCode.PDS_CONFIGURATION_ERROR);
+        }
+
+        Path parent = reportFile.getParent();
+        Path absolutePath = parent.toAbsolutePath();
+
+        return absolutePath.toString();
+    }
+
+    /**
+     * This method is used to rename the file back to the specified name in case the
+     * file did not end with .json.
+     *
+     * The reason for this method is that the Owasp Zap appends ".json" to the
+     * result file if we generate a report in json format. The PDS result.txt will
+     * then be called result.txt.json. Because of this behaviour the file will be
+     * renamed.
+     */
+    private void renameReportFileToOriginalNameIfNecessary() {
+        String specifiedReportFile = scanContext.getReportFile().toAbsolutePath().toFile().getAbsolutePath();
+        // If the Owasp Zap creates the file below, it will be renamed to the originally
+        // specified name
+        File owaspZapCreatedFile = new File(specifiedReportFile + ".json");
+        if (owaspZapCreatedFile.exists()) {
+            try {
+                Path owaspzapReport = Paths.get(specifiedReportFile + ".json");
+                Files.move(owaspzapReport, owaspzapReport.resolveSibling(scanContext.getReportFile().toAbsolutePath()), StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new ZapWrapperRuntimeException("For scan: " + scanContext.getContextName() + ". An error occurred renaming the report file", e,
+                        ZapWrapperExitCode.IO_ERROR);
+            }
+        }
+    }
+
+    private void cleanUpReplacerRules() throws ClientApiException {
+        if (scanContext.getSecHubWebScanConfiguration().getHeaders().isEmpty()) {
+            return;
+        }
+
+        List<HTTPHeaderConfiguration> httpHeaders = scanContext.getSecHubWebScanConfiguration().getHeaders().get();
+        for (HTTPHeaderConfiguration httpHeader : httpHeaders) {
+            if (httpHeader.getOnlyForUrls().isEmpty()) {
+                String description = httpHeader.getName();
+                clientApiFacade.removeReplacerRule(description);
+            } else {
+                for (String onlyForUrl : httpHeader.getOnlyForUrls().get()) {
+                    String description = onlyForUrl;
+                    clientApiFacade.removeReplacerRule(description);
+                }
+            }
+        }
+    }
+
+    private String urlEncodeUTF8(String stringToEncode) {
+        try {
+            return URLEncoder.encode(stringToEncode, StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            throw new IllegalStateException("This should not happen because we always use UTF-8: " + e);
+        }
+    }
+
+    class UserInformation {
```

Comment:
You could use a record type for this: https://dzone.com/articles/what-are-java-records.